### PR TITLE
support J21 in jenkins

### PR DESCRIPTION
**Pull request by cdietrich**

### CODE REVIEW

**Review by LorenzoBettini — APPROVED** [review]

**Comment by cdietrich** [comment]
Unfortunately this breaks the j11 build

**Comment by LorenzoBettini** [comment]
mh... why is it related?
However, I see we don't have a specification for jdk 11 in the tools in Jenkinsfile

**Comment by cdietrich** [comment]
Assume.assumeTrue("Active only on Java 21 and later", JavaRuntimeVersion.isJava21OrLater());

this guard does not seem to work

**Comment by LorenzoBettini** [comment]
This

```
11:26:42  [INFO] Command line:
11:26:42  	[/opt/tools/java/temurin/jdk-21/latest/bin/java
```

shows that Tycho is not using Java 11 to run the tests.

File: Jenkinsfile [hunk 1]
```diff
@@ -26,6 +26,7 @@ pipeline {
 
   tools {
      jdk "temurin-jdk17-latest"
```
Comment (LorenzoBettini) [inline Jenkinsfile:28]: Here we miss the specification for jdk11

**Comment by LorenzoBettini** [comment]
And this

```
11:19:26  [INFO] --- toolchains:3.1.0:toolchain (default) @ xtext-parent ---
11:19:26  [INFO] Required toolchain: jdk [ version='21' ]
11:19:26  [INFO] Found matching toolchain for type jdk: JDK[/opt/tools/java/temurin/jdk-21/latest]
```

shows that we either specify the wrong jdk version or it doesn't find a matching one

**Comment by LorenzoBettini** [comment]
I'll do some experiments on top of your branch.

We should check whether the whole thing used to work before this change...

**Comment by LorenzoBettini** [comment]
Either I miss something or from the main builds I cannot see any build with Java 11... wasn't that part of the timer-based builds (building with Java 11 and the olderst platform 4.23)?

**Comment by cdietrich** [comment]
grafik

**Comment by cdietrich** [comment]
maybe the daily java 11 buidls there broke too.
or it got lost.
i would have expected its still running.
but as i check fails only i have no idea

**Comment by LorenzoBettini** [comment]
@cdietrich I found the problem, and it's my fault O:)

we enable the `strict-release-jdk` profile (https://github.com/eclipse/xtext/blob/839911c9ebcc1fb7c18ae4563f03f6334b4e596b/pom.xml#L423) but then I had added another profile that gets activated automatically when the JDK is 21 (https://github.com/eclipse/xtext/blob/839911c9ebcc1fb7c18ae4563f03f6334b4e596b/pom.xml#L449) this other profile forces `<maven.compiler.release>21</maven.compiler.release>` and so the `strict-release-jdk` asks Maven to use Java 21 for the toolchain, voiding our goals for that profile.

I'll have to adjust a few things with those profiles, so that they won't disturb each other.

Maybe, the easiest and cleanest thing would be to have 2 profiles `strict-jdk-11` and `strict-jdk-21`, both activated explicitly, what do you think? This requires updating a few scripts of course.

**Comment by LorenzoBettini** [comment]
> maybe the daily java 11 buidls there broke too.
> or it got lost.

the one you put in the screenshot was started manually by me

**Comment by LorenzoBettini** [comment]
@cdietrich this should be superseded by #3105

**Pull request closed without merging**